src/logic/calculate.js

```diff
@@ -0,0 +1,130 @@
+import operate from './operate';
+import * as operations from './operationalConstants';
+
+const calculate = (calculator, buttonName) => {
+  const { total, next, operation } = calculator;
+  let returnValue = {
+    total,
+    next: null,
+    operation: null,
+  };
+
+  const resetCalculator = () => {
+    returnValue = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+  };
+
+  const assignSameValues = () => {
+    returnValue = {
+      total,
+      next,
+      operation,
+    };
+  };
+
+  if (buttonName === operations.CLEAR) {
+    resetCalculator();
+  } else if (Number.isNaN(+buttonName)) {
+    switch (buttonName) {
+      case operations.EQUAL:
+        if (total && next && operation) {
+          returnValue = {
+            total: operate(total, next, operation),
+            next: null,
+            operation: null,
+          };
+        } else {
+          assignSameValues();
+        }
+        break;
+      case operations.PERCENTAGE:
+        if (next) {
+          returnValue = {
+            total,
+            next: operate(next, '0.01', operations.MULTIPLY),
+            operation,
+          };
+        } else if (total) {
+          returnValue = {
+            total: operate(total, '0.01', operations.MULTIPLY),
+            next,
+            operation,
+          };
+        } else {
+          assignSameValues();
+        }
+        break;
```

Comment:
- [ ] Great job on the `calculate.js` file. But the instructions require that the modifier `operation (%)` should be deal with at the `operate.js`
![Screenshot from 2020-11-03 23-09-51](https://user-images.githubusercontent.com/30977845/98060988-b428d780-1e29-11eb-88d3-59f52d32c5c6.png)